dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventsExportControllerPostgresTest.java

```diff
@@ -0,0 +1,253 @@
+    de.getSharing().setOwner(owner);
+    manager.save(de, false);
+
+    programStage = createProgramStage('A', program);
+    programStage.setUid("pSllsjpfLH2");
+    program.getProgramStages().add(programStage);
+    ProgramStageDataElement programStageDataElement =
+        createProgramStageDataElement(programStage, de, 1, false);
+    manager.save(programStageDataElement);
+    programStage.setProgramStageDataElements(Sets.newHashSet(programStageDataElement));
+    manager.save(programStage);
+
+    dataValue = new EventDataValue();
+    dataValue.setDataElement(de.getUid());
+    dataValue.setStoredBy("user");
+    dataValue.setValue(DATA_ELEMENT_VALUE);
+  }
+
+  @Test
+  void shouldGetEventChangeLogWhenDataValueUpdated() {
+    Event event = event(enrollment(trackedEntity()));
+    event.getEventDataValues().add(dataValue);
+    manager.update(event);
+
+    String json =
+        """
+      {
+        "events": [
+          {
+            "event": "%s",
+            "status": "COMPLETED",
+            "program": "%s",
+            "programStage": "%s",
+            "enrollment": "%s",
+            "trackedEntity": "%s",
+            "orgUnit": "%s",
+            "occurredAt": "2023-01-10",
+            "scheduledAt": "2023-01-10",
+            "storedBy": "tracker",
+            "followUp": false,
+            "deleted": false,
+            "createdAt": "2018-01-20T10:44:03.222",
+            "createdAtClient": "2017-01-20T10:44:03.222",
+            "updatedAt": "2018-01-20T10:44:33.777",
+            "completedBy": "tracker",
+            "completedAt": "2023-01-20",
+            "notes": [],
+            "followup": false,
+            "geometry": null,
+            "dataValues": [
+              {
+                "dataElement": "%s",
+                "value": "%s"
+              }
+            ]
+          }
+        ]
+      }}
+        """
+            .formatted(
+                event.getUid(),
+                program.getUid(),
+                programStage.getUid(),
+                event.getEnrollment().getUid(),
+                event.getEnrollment().getTrackedEntity().getUid(),
+                event.getOrganisationUnit().getUid(),
+                event.getEventDataValues().iterator().next().getDataElement(),
+                NEW_DATA_ELEMENT_VALUE);
+
+    JsonWebMessage importResponse =
+        POST("/tracker?async=false&importStrategy=UPDATE", json)
+            .content(HttpStatus.OK)
+            .as(JsonWebMessage.class);
+    assertEquals(HttpStatus.OK.toString(), importResponse.getStatus());
+
+    JsonWebMessage changeLogResponse =
+        GET("/tracker/events/{id}/changelog", event.getUid())
+            .content(HttpStatus.OK)
+            .as(JsonWebMessage.class);
+    assertEquals(1, changeLogResponse.asList(JsonList.class).size());
+
+    JsonObject changeLogObject = changeLogResponse.asList(JsonList.class).get(0).asObject();
```

Comment:
We usually create something like https://github.com/dhis2/dhis2-core/blob/b69cfc3a2245cba91ca4641711ffbbd782ed8b89/dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/JsonEvent.java#L34

to make writing assertions easier.